clients/da-vinci-client/src/main/java/com/linkedin/davinci/ingestion/isolated/IsolatedIngestionServer.java

```diff
@@ -522,6 +529,72 @@ private void checkHeartbeatTimeout() {
     }
   }
 
+  void reportMetricsUpdateToMainProcess() {
+    try {
+      Map<CharSequence, Double> updatedMetricMap = new VeniceConcurrentHashMap<>();
+      if (getMetricsRepository() != null) {
+        getMetricsRepository().metrics().forEach((name, metric) -> {
+          if (metric != null) {
+            try {
+              // Best-effort to reduce metrics delta size sent from child process to main process.
+              Double originalValue = getMetricsMap().get(name);
+              Double newValue = metric.value();
+              if (originalValue == null || !originalValue.equals(newValue)) {
+                updatedMetricMap.put(name, newValue);
+              }
+              getMetricsMap().put(name, newValue);
+            } catch (Exception e) {
+              String exceptionLogMessage = "Encounter exception when retrieving value of metric: " + name;
+              if (!getRedundantExceptionFilter().isRedundantException(exceptionLogMessage)) {
+                LOGGER.error(exceptionLogMessage, e);
+              }
+            }
+          }
+        });
+      }
+
+      // Due to Netty message size limit, we will need to split the metric report into small batches if single report is
```

Comment:
BTW, I found this config:
```
    ch.pipeline().addLast(new HttpObjectAggregator(1024 * 1024));
```
In `MainIngestionReportChannelInitializer`, and this `1MB` is to control the max request size, so can you bump it up to see whether it would work without manual chunking?